Modules/xSharePoint/DSCResources/MSFT_xSPUserProfileSection/MSFT_xSPUserProfileSection.psm1

```diff
@@ -0,0 +1,148 @@
+function Get-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Collections.Hashtable])]
+    param
+    (
+        [parameter(Mandatory = $true)] [System.string] $Name ,
+        [parameter(Mandatory = $false)] [ValidateSet("Present","Absent")] [System.string ] $Ensure ,
+        [parameter(Mandatory = $true)] [System.string] $UserProfileService ,
+        [parameter(Mandatory = $false)] [System.string] $DisplayName ,
+        [parameter(Mandatory = $false)] [System.uint32] $DisplayOrder ,
+        [parameter(Mandatory = $false)] [System.Management.Automation.PSCredential] $InstallAccount
+    )
+
+    Write-Verbose -Message "Getting user profile service application $Name"
+
+    $result = Invoke-xSharePointCommand -Credential $InstallAccount -Arguments $PSBoundParameters -ScriptBlock {
+        $params = $args[0]
+        
+        $upsa = Get-SPServiceApplication -Name $params.UserProfileService -ErrorAction SilentlyContinue 
+        if ($null -eq $upsa) { 
+            return $null 
+        }
+        $caURL = (Get-SpWebApplication  -IncludeCentralAdministration | ?{$_.IsAdministrationWebApplication -eq $true }).Url
+        $context = Get-SPServiceContext -Site $caURL 
+        $userProfileConfigManager  = new-object Microsoft.Office.Server.UserProfiles.UserProfileConfigManager($context)
+        
+        $userProfileSubTypeManager = Get-xSharePointUserProfileSubTypeManager $context
+        $userProfileSubType = $userProfileSubTypeManager.GetProfileSubtype("UserProfile")
+        
+        $userProfileProperty = $userProfileSubType.Properties.GetSectionByName($params.Name) 
+        if($userProfileProperty -eq $null){
+            return $null 
+        }
+        
+        
+        return @{
+            Name = $userProfileProperty.Name 
```

Comment:
Should probably also return ensure so I don't have to infer if the section is present or absent.